python/triton/experimental/gluon/language/amd/_layouts.py

```diff
@@ -65,9 +69,69 @@ def _to_ir(self, builder):
 
     def mangle(self) -> str:
 
-        def stringify(x):
-            if x is None:
-                return ""
-            return "_".join(map(str, x))
+        return f"MFMA_{self.version}_{_stringify(self.instr_shape)}_{self.transposed}_{_stringify(self.warps_per_cta)}_{_stringify(self.tiles_per_warp)}_{self.elem_type}_{_stringify(self.ctas_per_cga)}_{_stringify(self.cta_split_num)}_{_stringify(self.cta_order)}_MFMA"
+
 
-        return f"MFMA_{self.version}_{stringify(self.instr_shape)}_{self.transposed}_{stringify(self.warps_per_cta)}_{stringify(self.tiles_per_warp)}_{self.elem_type}_{stringify(self.ctas_per_cga)}_{stringify(self.cta_split_num)}_{stringify(self.cta_order)}_MFMA"
+@dataclass(frozen=True)
+class PaddedSharedLayout(DistributedLayout):
+    """
+    Represents a layout for the access to AMD shared memory. Compared to SwizzledSharedLayout,
+    it uses padding to avoid shared memory bank conflicts.
+    One concrete example, using `eM` to mean tensor elements and `pN` to mean padding and interval-padding pairs are [2:+1, 4:+2]
+
+    #ttg.padded_shared<[2:+1, 4:+2]>
+    [e0, e1, p0,
+     e2, e3, p1, p2, p3,
+     e4, e5, p4,
+     e6, e7, p5, p6, p7,
+     ...]
+
+    Args:
+        intervals: List[int], list of intervals in the interval-padding pairs, e.g [2, 4] in the exmaple above.
+        paddings: List[int], list of paddings in the interval-padding pairs, e.g [1, 2] in the exmaple above.
+        order: List[int], order of logical tensor dimensions; fastest-varying first.
+        ctas_per_cga (Optional[List[int]]): CTAs per CGA grouping.
+        cta_split_num (Optional[List[int]]): Split factors for CTAs.
+        cta_order (Optional[List[int]]): CTA ordering.
+    """
+    intervals: List[int]
+    paddings: List[int]
```

Comment:
Instead of doing this way, I think a `List[Tuple[int, int]]` might be better interface for developers.